core/src/trezor/crypto/rlp.py

```diff
@@ -1,53 +1,94 @@
-def int_to_bytes(x: int) -> bytes:
-    if x == 0:
-        return b""
-    r = bytearray()
-    while x:
-        r.append(x % 256)
-        x //= 256
-    return bytes(reversed(r))
-
-
-def encode_length(l: int, is_list: bool) -> bytes:
-    offset = 0xC0 if is_list else 0x80
-    if l < 56:
-        return bytes([l + offset])
-    elif l < 256 ** 8:
-        bl = int_to_bytes(l)
-        return bytes([len(bl) + offset + 55]) + bl
+from micropython import const
+
+if False:
+    from typing import Union
+    from trezor.utils import Writer
+
+    RLPItem = Union["RLPItem", bytes, int]
+
+
+STRING_HEADER_BYTE = const(0x80)
+LIST_HEADER_BYTE = const(0xC0)
+
+
+def _byte_size(x: int) -> int:
+    if x < 0:
+        raise ValueError  # only unsigned ints are supported
+    for exp in range(64):
+        if x < 0xFF ** exp:
```

Comment:
This seems weird. Maybe you meant:

```suggestion
        if x < 0x100 ** exp:
```

Reply:
> Also this function returns 0 for x=0. Not sure if that is the intention.

This is intentional, yes

But good point about the equality test. I'll need to add an edge value to the testcases first as it's apparently not covered.

Reply:
3d4d186